setup.py

```diff
@@ -869,6 +875,148 @@ def prefix(directory, files):
+                    os.path.join(sz3_dir, "include", "SZ3"),
+                    os.path.join(sz3_dir, "include", "SZ3", "api"),
+                    os.path.join(sz3_dir, "include", "SZ3", "compressor"),
+                    os.path.join(sz3_dir, "include", "SZ3", "encoder"),
+                    os.path.join(sz3_dir, "include", "SZ3", "frontend"),
+                    os.path.join(sz3_dir, "include", "SZ3", "lossless"),
+                    os.path.join(sz3_dir, "include", "SZ3", "predictor"),
+                    os.path.join(sz3_dir, "include", "SZ3", "preprocessor"),
+                    os.path.join(sz3_dir, "include", "SZ3", "quantizer"),
+                    os.path.join(sz3_dir, "include", "SZ3", "utils"),
+                    os.path.join(sz3_dir, "include", "SZ3", "utils", "inih"),
+                    os.path.join(sz3_dir, "include", "SZ3", "utils", "ska_hash"),
+                    ]
+if sys.platform.startswith('darwin'):
+    # provide dummy omp.h
+    sz3_include_dirs.append(os.path.join("src", "SZ3_extra", "darwin"))
+
+sz3_hdf5_dir = os.path.join(sz3_dir, "tools", "H5Z-SZ3")
+
+# I have issues with the BMI on linux and Mac because of compiling Zstd separately
+
+if sys.platform.startswith('darwin'):
+    HDF5PLUGIN_ZSTD_FROM_BLOSC = False
+else:
+    HDF5PLUGIN_ZSTD_FROM_BLOSC = True
+
+if not HDF5PLUGIN_ZSTD_FROM_BLOSC:
+    sz3_zstd_cmake_list = [
+        './common/entropy_common.c',
+        './common/pool.c',
+        './common/threading.c',
+        './common/debug.c',
+        './common/xxhash.c',
+        './common/fse_decompress.c',
+        './common/zstd_common.c',
+        './common/error_private.c',
+        './compress/zstd_ldm.c',
+        './compress/zstd_lazy.c',
+        './compress/huf_compress.c',
+        './compress/zstd_opt.c',
+        './compress/zstd_double_fast.c',
+        './compress/zstd_compress.c',
+        './compress/zstd_compress_superblock.c',
+        './compress/zstd_compress_sequences.c',
+        './compress/zstd_compress_literals.c',
+        './compress/zstd_fast.c',
+        './compress/fse_compress.c',
+        './compress/zstdmt_compress.c',
+        './compress/hist.c',
+        './decompress/zstd_decompress.c',
+        './decompress/huf_decompress.c',
+        './decompress/zstd_ddict.c',
+        './decompress/zstd_decompress.c',
+        './decompress/zstd_decompress_block.c',
+        './deprecated/zbuff_common.c',
+        './deprecated/zbuff_compress.c',
+        './deprecated/zbuff_decompress.c',
+        './legacy/zstd_v05.c',
+        './legacy/zstd_v04.c',
+        './legacy/zstd_v06.c',
+        './legacy/zstd_v07.c',
+        './legacy/zstd_v03.c',
+        './legacy/zstd_v02.c',
+        './legacy/zstd_v01.c',
+        './dictBuilder/cover.c',
+        './dictBuilder/divsufsort.c',
+        './dictBuilder/zdict.c',
+        './dictBuilder/fastcover.c',
+        ]
+
+    zstd_extra_objects = []
+    zstd_define_macros = []
+    zstd_sources = []
+    zstd_depends = []
```

Comment:
That's risky if we add more filters below.
I reckon we should improve the reuse of zstd, but in the mean time, what about doing it the same way as for the other filters? : defining `extra_objects`, `define_macros`,...; appending to those and using them in `HDF5PluginExtension`